### PR TITLE
Rpmbuildfix

### PR DESCRIPTION
**Pull request by gbuloz**
Patches to fix the broken RPM build process, simplify it, and update the changelog

### CODE REVIEW

File: control/rpmmacros.in [hunk 1]
```diff
@@ -1,4 +1,3 @@
-%_topdir	%(echo $BUILDDIR)
```
Comment (AlexanderAmelkin) [inline control/rpmmacros.in]: Why bother and change this line here if you remove the whole file in another commit?

File: Makefile.am [hunk 2]
```diff
@@ -93,14 +93,8 @@ SRCDIR   = ${shell cd $(top_srcdir) ; pwd | sed -e 's,^[^:\\/]:[\\/],/,'}
...
-rpm: control/ipmitool.spec dist
-	mkdir -p $(RPMDIR)/{BUILD,RPMS,SRPMS,SOURCES,SPECS,tmp}
-	cp control/ipmitool.spec $(RPMDIR)/SPECS
-	cp $(distdir).tar.gz $(RPMDIR)/SOURCES
-	CONFIGDIR=$(BUILDDIR)/control \
-	  @RPMBUILD@ -ba --define "_topdir $(RPMDIR)" \
-	  --rcfile $(SRCDIR)/control/rpmrc \
-	  $(RPMDIR)/SPECS/ipmitool.spec
+rpm: dist
+	@RPMBUILD@ -ta $(distdir).tar.gz
```
Comment (AlexanderAmelkin) [inline Makefile.am]: What's the urge to build in ~/rpmbuild instead of where it was building? Was there any problem with that? If so, it needs an explanation in the commit log.

File: include/ipmitool/Makefile.am [hunk 3]
```diff
@@ -39,4 +39,4 @@ noinst_HEADERS = log.h bswap.h hpm2.h helper.h ipmi.h ipmi_cc.h ipmi_intf.h \
 	ipmi_fwum.h ipmi_main.h ipmi_tsol.h ipmi_firewall.h \
 	ipmi_kontronoem.h ipmi_ekanalyzer.h ipmi_gendev.h ipmi_ime.h \
 	ipmi_delloem.h ipmi_dcmi.h ipmi_vita.h ipmi_sel_supermicro.h \
-	ipmi_cfgp.h ipmi_lanp6.h ipmi_quantaoem.h
+	ipmi_cfgp.h ipmi_lanp6.h ipmi_quantaoem.h ipmi_time.h
```
Comment (AlexanderAmelkin) [inline include/ipmitool/Makefile.am]: Looks like this change need to be in a separate PR as it is not directly related to RPM.

File: control/ipmitool.spec.in [hunk 4]
```diff
@@ -64,6 +64,626 @@ fi
 
 
 %changelog
+* Sat Oct 8 2016 <pere@hungry.com>  1.8.18-1
```
[the remark this answers is not in the document]
Comment (gbuloz) [reply control/ipmitool.spec.in]: <pere@hungry.com> is Petter Reinholdtsen. See file AUTHORS.
On branch master, the last entry in ./control/ipmitool.spec.in has "* Wed Feb 25 2009 <pere@hungry.com>  1.8.11-1"
I've just updated this file to have it at the same level than ./Changelog and also used pere@hungry.com for each added version entry. I need an email for each version in this file; which one should I use instead ?
The Changelog file ends at "version 1.8.18 2016-10-08". Yes, clearly it is not up-to-date. Is it still maintained ? Should it be updated from git log in some way ?

File: control/rpmmacros.in [hunk 5]
```diff
@@ -1,11 +0,0 @@
...
-%_specdir       %{_sourcedir}
-%_tmppath	%{_topdir}/tmp
-%_builddir	%{_topdir}/BUILD
-%_buildroot	%{_topdir}/%{_tmppath}/%{name}-%{version}-root
-%_rpmdir	%{_topdir}/RPMS
-%_srcrpmdir	%{_topdir}/SRPMS
-%_distro	@DISTRO@
-%_rpmfilename	%%{NAME}-%%{VERSION}-%%{RELEASE}.%%{ARCH}.rpm
-%packager	jmank@hp.com
-%distribution	Sourceforge Build
```
[the remark this answers is not in the document]
Comment (gbuloz) [reply control/rpmmacros.in]: Sorry, for the distribution name, I'll add this line to the spec file :
Distribution: GitHub Build

File: control/rpmmacros.in (hunk 1 above)
Comment (gbuloz) [reply control/rpmmacros.in]: Your're right, but this is only because I've split my changes into several commits. The first one fixes the RPM build and contains this change. Then the second one is to simplify the RPM build process and removes this file.

File: include/ipmitool/Makefile.am (hunk 3 above)
Comment (gbuloz) [reply include/ipmitool/Makefile.am]: Yes, not directly related to RPM, but I saw the problem when attempting to build an RPM. Can this problem be seen in another case ?

File: Makefile.am (hunk 2 above)
Comment (gbuloz) [reply Makefile.am]: No urge, but now once you've run "make dist" to create a source archive you can give it to somebody else and have it run simply "rpmbuild -tb ipmitool-*.tar.gz" to build an RPM. This is also the standard location to build an RPM as a regular user. And this also makes the Makefile much simple (only 1 line instead of 8 for the "rpm" target)
Sorry, I've forgotten to explain that in the commit message.

**Comment by gbuloz** [comment]
Sorry I'm quite new to GitHub and would like to know how I'm expected to rework my pull request. I've been unable to find the answer.
Should I rebase, change my commits, and maybe insert some commits between my commits ?
Should I rework by applying new commits ?
Should I for again from the master and create another feature branch ?
Note : I've merged master to my rpmbuildfix branch today but I'm not sure this was a good idea

File: control/ipmitool.spec.in (hunk 4 above)
[the remark this answers is not in the document]
Comment (AlexanderAmelkin) [reply control/ipmitool.spec.in]: I believe it only makes sene to update the Changelog file as well as changelog in the spec when code is freezed and a new release is being prepared. Otherwise each commit would require updating them.

File: Makefile.am (hunk 2 above)
Comment (AlexanderAmelkin) [reply Makefile.am]: I see. Please do explain that in the commit message.

**Comment by AlexanderAmelkin** [comment]
Please either add new commits to the same branch (they will automatically appear here) or interactively rebase to master and change each commit as needed.

Please see https://github.com/ipmitool/ipmitool/wiki/How-to-contribute#2-contribute-the-changes

File: control/ipmitool.spec.in (hunk 4 above)
[the remark this answers is not in the document]
Comment (gbuloz) [reply control/ipmitool.spec.in]: Which email address do you recommend me to put for the entries I've added to the changelog in the spec file ?

File: Makefile.am (hunk 2 above)
Comment (gbuloz) [reply Makefile.am]: Good news : I've been able to simplify things and keep the build directory unchanged : under ./rpmbuild for "make rpm", and ~/rpmbuild/ for "rpmbuild -tb ipmitool-*.tar.gz"

File: include/ipmitool/Makefile.am (hunk 3 above)
Comment (gbuloz) [reply include/ipmitool/Makefile.am]: I've put this in a separate commit

File: control/ipmitool.spec.in [hunk 6]
```diff
@@ -359,7 +983,7 @@ fi
  - Fix SEL event decoding for generic events
  - Handle empty SEL gracefully when doing "sel list"
  - Fix sdr handling of sensors that do not return a reading
- - Fix for CSV display of sensor readings/units from Fredrik �hrn
+ - Fix for CSV display of sensor readings/units from Fredrik ohrn
```
Comment (AlexanderAmelkin) [inline control/ipmitool.spec.in:986]: I don't think so. You should be using a UTF-8 compatible locale and editor, because that was `Fredrik Öhrn` and should stay so.

File: control/ipmitool.spec.in [hunk 7]
```diff
@@ -1,10 +1,13 @@
+%define debug_package %{nil}
+
 Name:         ipmitool
 Summary:      ipmitool - Utility for IPMI control
 Version:      @VERSION@
 Release:      @RPM_RELEASE@%{?_distro:.%{_distro}}
 License:      BSD
 Group:        Utilities
 Packager:     Jim Mankovich <jmank@hp.com>
```
Comment (AlexanderAmelkin) [inline control/ipmitool.spec.in:9]: Well, while we're at it, this spec file is just a template. I don't think it needs to contain the name of Jim because the resulting RPM package will definitely NOT be packaged by him. I think there must be some macro here like those `@VERSION@` and `@RPM_RELEASE@` above. I just don't know which.

File: control/rpmrc [hunk 8]
```diff
@@ -1,8 +0,0 @@
-macrofiles: /usr/lib/rpm/macros:/usr/lib/rpm/%{_target}/macros:/etc/rpm/macros.specspo:/etc/rpm/macros:/etc/rpm/%{_target}/macros:%(echo $CONFIGDIR)/rpmmacros
-
-buildarchtranslate: athlon: i386
-buildarchtranslate: i686: i386
-buildarchtranslate: i586: i386
-buildarchtranslate: i486: i386
-buildarchtranslate: i386: i386
```
Comment (AlexanderAmelkin) [inline control/rpmrc:7]: Did we really need to remove those translations? Didn't we break anything with that?

File: control/ipmitool.spec.in (hunk 4 above)
[the remark this answers is not in the document]
Comment (gbuloz) [reply control/ipmitool.spec.in:68]: OK, I've put my email for the releases I've added to the changelog.

File: control/ipmitool.spec.in (hunk 6 above)
Comment (gbuloz) [reply control/ipmitool.spec.in:986]: OK, fixed. I was using UTF8 but something went wrong when I imported the text from the Changelog file.

File: control/ipmitool.spec.in (hunk 7 above)
Comment (gbuloz) [reply control/ipmitool.spec.in]: To make it simple, as the Packager info is not mandatory, I've put a commented example  : 
#Packager:    PackagerName <packager@example.com>

File: control/rpmrc (hunk 8 above)
Comment (gbuloz) [reply control/rpmrc:7]: All these translations are already defined in the default /usr/lib/rpm/rpmrc used by rpmbuild, also having some other translations for 32bit and 64bit are also in this file (osfmach3, geode, pentium4, pentium3, amd64, ia32e,...). So it seems to me better to use this file that is much more up-to-date.
Some distributions such as Fedora are building all their packages using i386 options for all 32bit intel, but use i686 as architecture in the package name. This is done by the /usr/lib/rpm/redhat/rpmrc file. I think it's best to follow this when building a package for such distributions to have the same settings than other packages.
And in this case if forcing i386 is required for any reason, this can be done by passing--rpmrc /usr/lib/rpm/rpmrc to rpmbuild, or by adding a file /etc/rpmrc or ~/.rpmrc with the required single buildarchtranslate: line.
That's why I think in any case we can remove our control/rpmrc file.
